src/WorkOS.net/Services/Webhooks/WebhookService.cs

```diff
@@ -0,0 +1,219 @@
+namespace WorkOS
+{
+    using System;
+    using System.Security.Cryptography;
+    using System.Text;
+    using Newtonsoft.Json;
+
+    public class WebhookService
+    {
+        private const int DefaultTimeTolerance = 300;
+
+        public static bool ConstantTimeAreEqual(byte[] a, byte[] b)
+        {
+            int i = a.Length;
+            if (i != b.Length)
+            {
+                return false;
+            }
+
+            int cmp = 0;
+            while (i != 0)
+            {
+                --i;
+                cmp |= a[i] ^ b[i];
+            }
+
+            return cmp == 0;
+        }
+
+        /// <summary>
+        /// Parses a JSON string from a webhook, verifies and deseralizes into a Webhook Object.
+        /// </summary>
+        /// <param name="json">The JSON string to parse.</param>
+        /// <param name="signatureHeader">
+        /// The value of the header from the webhook request.
+        /// </param>
+        /// <param name="secret">The webhook endpoint's signing secret.</param>
+        /// <param name="tolerance">The time tolerance, in seconds.</param>
+        /// <returns>The deserialized JSON.</returns>
+        public Webhook ConstructEvent(
+            string json,
+            string signatureHeader,
+            string secret,
+            long tolerance)
+        {
+            if (this.VerifyHeader(json, signatureHeader, secret, tolerance))
+            {
+                Webhook webhookVerified = JsonConvert.DeserializeObject<Webhook>(json);
+                return webhookVerified;
+            }
+            else
+            {
+                throw new Exception("Unable to verify Webhook");
+            }
+        }
+
+        /// <summary>
+        /// Verify Header.
+        /// </summary>
+        /// <param name="payload">
+        /// json payload.
```

Comment:
Can we use proper sentence casing for the parameters in the doc comments?

```suggestion
        /// The JSON payload.
```

Also, for `param` declarations, it's generally preferred to include the tags on the same line as the content to cut down on the vertical space used up:

```cs
/// <param name="payload">The JSON payload.</param>
```

Reply:
addressed sentence casing and cleaned up parameters tags to one line each